tests/MixinAnnotationTest.php

```diff
@@ -243,10 +243,10 @@ public function __construct ($var) {
                         }
 
                         /**
-                         * @psalm-return T
+                         * @psalm-return self<T>
```

Comment:
Any reason why you can't just create a new test?

Reply:
figured it'd be a lot of boilerplate, happy to do so :) I'll push it up in a sec